portal-impl/src/com/liferay/portal/util/PropsUtil.java

```diff
@@ -383,29 +393,27 @@ private static String _getLibDir(Class<?> clazz) {
 				portalShieldedContainerLibDirProperty += StringPool.SLASH;
 			}
 
-			portalShieldedContainerLibDir =
-				portalShieldedContainerLibDirProperty;
-		}
+			SystemProperties.set(
+				PropsKeys.LIFERAY_SHIELDED_CONTAINER_LIB_PORTAL_DIR,
+				portalShieldedContainerLibDirProperty);
 
-		SystemProperties.set(
-			PropsKeys.LIFERAY_SHIELDED_CONTAINER_LIB_PORTAL_DIR,
-			portalShieldedContainerLibDir);
+			// Portal web directory
 
-		// Portal web directory
+			String portalWebDir = portalShieldedContainerLibDirProperty;
 
-		String portalWebDir = portalShieldedContainerLibDir;
+			if (portalWebDir.endsWith("/WEB-INF/shielded-container-lib/")) {
+				portalWebDir = portalWebDir.substring(
+					0, portalWebDir.length() - 31);
+			}
 
-		if (portalWebDir.endsWith("/WEB-INF/shielded-container-lib/")) {
-			portalWebDir = portalWebDir.substring(
-				0, portalWebDir.length() - 31);
-		}
+			if (_log.isDebugEnabled()) {
+				_log.debug("Portal web directory " + portalWebDir);
+			}
 
-		if (_log.isDebugEnabled()) {
-			_log.debug("Portal web directory " + portalWebDir);
+			SystemProperties.set(
+				PropsKeys.LIFERAY_WEB_PORTAL_DIR, portalWebDir);
```

Comment:
only set LIFERAY_WEB_PORTAL_DIR when portalShieldedContainerLibDirProperty is set